lecture_1/lecture_1_passed/passed.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+type passedAPIResponse struct{
+	Name string
+	Age int
+	Passed bool
+	Skills []string
+}
+
+
+
+const passedUrl = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+func writingToFile(f *os.File, decodedContent []passedAPIResponse){
+	for i := range decodedContent{
+		if decodedContent[i].Passed{
+			var found = false
+			//var knownSkills []string
+			var knownSkillText = strings.Join(decodedContent[i].Skills, ", ")
+			for _,val := range decodedContent[i].Skills{
+				//knownSkills = append(knownSkills, val)
+
+				if val == "Java" || val == "Go"{
+					found = true
+				}
+			}
```

Comment:
Sometimes, it's worth to sacrifice performance for readability. In this case, you could write a `contains` function and check both skills separately. This does mean that you go over your array twice, but it still runs in `O(n)`.
```go
if contains(val.Skills, "Java") || contains(val.Skills, "Go") {
    // do something
}
```